homeworks/09-js-advanced-2/script.js

```
@@ -0,0 +1,108 @@
+class Serializable {
+  serialize() {
+    const props = Object.getOwnPropertyNames(this);
+    for (let i = 0; i < props.length; i++) {
+      let propsValue = this[props[i]];
+      if (
+        propsValue == Infinity ||
+        propsValue == -Infinity ||
+        propsValue == NaN ||
+        propsValue == -0
+      ) {
+        propsValue = '"' + `${propsValue}` + '"';
+      }
+    }
+    return JSON.stringify(this);
+  }
+  ckeckDate(data) {
+    return new Date(data) !== "Invalid Date" && !isNaN(new Date(data));
+  }
+  wakeFrom(data) {
+    if (data) {
+      let item = JSON.parse(data);
+      const props = Object.getOwnPropertyNames(this);
+      for (let keys in item) {
+        let found = false;
+        for (let i = 0; i < props.length; i++) {
+          let propValue = item[props[i]];
+          let itemType = typeof propValue;
+          if (
+            ![
+              "null",
+              "object",
+              "array",
+              "number",
+              "string",
+              "Date",
+              "undefined",
```

undefined is not supported and should throw an Error